### PR TITLE
fix(console): reduce refresh frequency in preview

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Pass form data to preview box only when form is dirty, otherwise, use remote data directly.

This can reduce "postMessage" frequency to avoid unexpected refresh.

## Testing

Local tested, change one form value, and count postMessage calls:

before:

截屏2022-05-25 下午5 40 27

after reduce:

image

### CODE REVIEW

**Review by gao-sun — APPROVED** [review]
let's add testing info before merging

File: packages/console/src/pages/SignInExperience/index.tsx [hunk 1]
```diff
@@ -41,11 +41,19 @@ const SignInExperience = () => {
...
   const formData = watch();
 
+  const previewConfigs = useMemo(() => {
+    if (!isDirty) {
+      return data;
+    }
+
+    return signInExperienceParser.toRemoteModel(formData);
+  }, [formData, isDirty, data]);
+
```
Comment (demonzoo) [inline packages/console/src/pages/SignInExperience/index.tsx:56]: Maybe we should extract it into a hook and re-use it in both GuideModal and SignInExperience page

Comment (demonzoo) [reply packages/console/src/pages/SignInExperience/index.tsx:56]: Hmm, maybe even re-use the form? It seems everything has two copies

Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/index.tsx:51]: They two have much difference actually.

I refactored this hook logic, PTAL

**Pull request merged**